src/librustdoc/html/render/print_item.rs

```diff
@@ -1249,6 +1250,9 @@ fn item_type_alias(w: &mut Buffer, cx: &mut Context<'_>, it: &clean::Item, t: &c
         match inner_type {
             clean::TypeAliasInnerType::Enum { variants, is_non_exhaustive } => {
                 let variants_iter = || variants.iter().filter(|i| !i.is_stripped());
+                let ty = cx.tcx().type_of(it.def_id().unwrap()).instantiate_identity();
+                let enum_def_id = ty.ty_adt_id().unwrap();
```

Comment:
I wish we didn't need to do this and could just store the `DefId` of the inner item inside `TypeAliasInnerType` when `clean[ing]_ty_alias_inner_type` but that would increase the size of the type ... so your code is fine.

Reply:
Unfortunately yes. We could maybe add a helper method later on if we need to do it once again though,